os/fs/vfs/fs_read.c

```diff
@@ -111,23 +117,16 @@ ssize_t file_read(FAR struct file *filep, FAR void *buf, size_t nbytes)
 	 * method?
 	 */
 
-	else if (inode && inode->u.i_ops && inode->u.i_ops->read) {
+	else if (inode != NULL && inode->u.i_ops && inode->u.i_ops->read) {
 		/* Yes.. then let it perform the read.  NOTE that for the case of the
 		 * mountpoint, we depend on the read methods being identical in
 		 * signature and position in the operations vtable.
 		 */
 
-		ret = (int)inode->u.i_ops->read(filep, (char *)buf, (size_t)nbytes);
-	}
-
-	/* If an error occurred, set errno and return -1 (ERROR) */
-
-	if (ret < 0) {
-		set_errno(-ret);
```

Comment:
As Taejun said, it seems tgat if this set_errno() is removed, we need to add the set_errno() to every place on which file_read is called..
Are there some reasons that this removal is better?

Reply:
1. I sync the Nuttx code. For future porting, following the Nuttx code is a good choice.
2. The errno should be set just before the function returns. errno can be polluted by other functions.